src/evaluator/operators.ts

```diff
@@ -1,185 +1,192 @@
-import type {OpCall} from '../nodeTypes'
-import {
-  FALSE_VALUE,
-  fromDateTime,
-  fromJS,
-  fromNumber,
-  fromString,
-  NULL_VALUE,
-  StreamValue,
-  TRUE_VALUE,
-  type Value,
-} from '../values'
-import {isEqual} from './equality'
-import {
-  gatherText,
-  matchAnalyzePattern,
-  matchText,
-  matchTokenize,
-  type Pattern,
-  type Token,
-} from './matching'
-import {partialCompare} from './ordering'
-
-type GroqOperatorFn = (left: Value, right: Value) => Value | PromiseLike<Value>
-
-export const operators: {[key in OpCall]: GroqOperatorFn} = {
-  '==': function eq(left, right) {
-    return isEqual(left, right) ? TRUE_VALUE : FALSE_VALUE
-  },
-
-  '!=': function neq(left, right) {
-    return isEqual(left, right) ? FALSE_VALUE : TRUE_VALUE
-  },
-
-  '>': function gt(left, right) {
-    if (left.type === 'stream' || right.type === 'stream') return NULL_VALUE
-    const result = partialCompare(left.data, right.data)
-
-    if (result === null) {
-      return NULL_VALUE
-    }
-    return result > 0 ? TRUE_VALUE : FALSE_VALUE
-  },
+/* eslint-disable complexity */
+/* eslint-disable max-statements */
 
-  '>=': function gte(left, right) {
-    if (left.type === 'stream' || right.type === 'stream') return NULL_VALUE
-    const result = partialCompare(left.data, right.data)
+import type {OpCallNode} from '../nodeTypes'
+import {isEqual} from './equality'
+import {evaluate, isIso8601, isIterable} from './evaluate'
+import {matchAnalyzePattern, matchText, matchTokenize} from './matching'
+import {compare} from './ordering'
+import type {Context} from './types'
 
-    if (result === null) {
-      return NULL_VALUE
-    }
-    return result >= 0 ? TRUE_VALUE : FALSE_VALUE
-  },
+interface EvaluateOpCallOptions extends Context {
+  node: OpCallNode
+}
```

Comment:
I don't quite see the need for this. Why not use multiple parameters? Isn't the current restructuring just adding lots of unnecessary allocations anyway?